### PR TITLE
feat: Integrate amplitude SDK along other analytics SDKs

### PR DESCRIPTION
**Pull request by mudit-postman**
Base on the discussion on #202, this PR now integrates Amplitude SDK along with the other SKDs to support backward compatability

### CODE REVIEW

File: analytics/client.go [hunk 1]
```diff
@@ -29,33 +30,15 @@ type clientImpl struct {
 
 	// TODO: Remove Mixpanel once we've confirmed that Segment is working.
 	mixpanelClient mixpanel.Mixpanel
+
+	// The internal client used to send events to Amplitude.
+	amplitudeClient amplitude.Client
+	// App info from which event is sent
+	amplitudeAppInfo amplitude.EventOptions
```
Comment (liujed) [inline analytics/client.go:37]: This comment is hard to understand. I have a suggestion for improvement below. However, `amplitudeClient` really belong together `amplitudeAppInfo`, so we'd be better served by having an abstraction that wraps them both and injects `amplitudeAppInfo` into all reported tracking events. **Non-blocking.**

```suggestion
	// This is included in all tracking events reported to Amplitude.
	amplitudeAppInfo amplitude.EventOptions
```

File: analytics/client.go [hunk 2]
```diff
@@ -149,6 +167,69 @@ func newMixpanelClient(config Config) (mixpanel.Mixpanel, error) {
 	return mixpanel.New(config.MixpanelToken, mixpanelURL), nil
 }
 
+func newAmplitudeClient(config Config) (amplitude.Client, amplitude.EventOptions, error) {
+	if !config.IsAmplitudeEnabled || config.AmplitudeConfig == (AmplitudeConfig{}) {
```
Comment (liujed) [inline analytics/client.go:171]: Why provide two different ways of disabling the Amplitude client? If the configuration has explicitly specified that Amplitude is enabled, shouldn't the absence of a configuration be an error? **Blocking.**

File: analytics/client.go [hunk 3]
```diff
@@ -205,3 +295,42 @@ func (d disabledLogger) Logf(format string, args ...any) {
 func (d disabledLogger) Errorf(format string, args ...interface{}) {
 	// Do nothing.
 }
+
+// A custom segment logger that logs to glog.
+type amplitudeLogger struct{}
```
Comment (liujed) [inline analytics/client.go:300]: This file is starting to feel a bit cluttered. We should separate the code for Mixpanel, Segment, and Amplitude out into separate files. **Non-blocking.**

**Review by liujed — APPROVED** [review]
Looks good. A few minor comments. Please address at least the one blocking comment before merging.

File: analytics/client.go (hunk 3 above)
Comment (mudit-postman) [reply analytics/client.go:304]: I have not done it cause we might be removing the segment and Mixpanel SDKs when we are sure that Amplitude is working fine and all the metrics are passed to Amplitude.

**Pull request merged**